cmd/test-server/main.go

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/apiserver-network-proxy/pkg/util"
+)
+
+func main() {
+	testServer := &TestServer{}
+	o := newTestServerRunOptions()
+	command := newTestServerCommand(testServer, o)
+	flags := command.Flags()
+	flags.AddFlagSet(o.Flags())
+	local := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	klog.InitFlags(local)
+	local.VisitAll(func(fl *flag.Flag) {
+		fl.Name = util.Normalize(fl.Name)
+		flags.AddGoFlag(fl)
+	})
+	if err := command.Execute(); err != nil {
+		klog.Errorf("error: %v\n", err)
+		klog.Flush()
+		os.Exit(1)
+	}
+}
+
+type TestServerRunOptions struct {
+	// Certificate setup for securing communication to the "client" i.e. the Kube API Server.
```

Comment:
Are these variables used anywhere? Should we just drop them for now since they're unused?

Reply:
No good point. I'll drop them for now. At some point we may want to test things about https inside the tunnel, but that day, is not today.